src/interfaces/records-read.ts

```diff
@@ -57,6 +67,32 @@ export class RecordsRead extends Message<RecordsReadMessage> {
     return new RecordsRead(message);
   }
 
+  /**
+   * Creates a Records interface Filter using either the `recordId` or `protocol` & `protocolPath` of the incoming Message.
+   * Also filters on an optional `parentId` for `protocolPath` based reads
+   * @param descriptor message descriptor with optional properties `recordId`, `protocol` and `protocolPath`
+   *
+   * @returns {Filter} with a Records interface as well as the appropriate filter params
+   * @throws {DwnError} if `recordId` or `protocol` and `protocolPath` are missing
+   */
+  public static createFilter(descriptor: Partial<Omit<RecordsReadDescriptor, 'method'>>): Filter {
+    const filter: Filter = { interface: DwnInterfaceName.Records, isLatestBaseState: true };
+    const { recordId, protocol, protocolPath, parentId } = descriptor;
+    if (recordId !== undefined) {
+      return { ...filter, recordId };
+    } else if (protocol !== undefined && protocolPath !== undefined) {
+      if (parentId !== undefined) {
+        filter.parentId = parentId;
+      }
```

Comment:
What is the use case for supporting `parentId`?

Reply:
Was thinking this would be useful if you have a protocol with multiple children paths that you may want to read a record from.

Say you have a game protocol with multiple `game` paths and you'd like to read the latest `game/score` from a particular game, you would need that game's `parentId`. You could read the score of the latest updated game without it, or target a specific game.

As I mentioned in the Issue discussion https://github.com/TBD54566975/dwn-sdk-js/issues/468#issuecomment-1682174577, I'd be open to removing this if it doesn't fit the intended purpose of this change.

Reply:
Let's remove it unless @csuwildcat have other thoughts. My understanding is that the whole point of this feature is to avoid needing to learning a CID that requires prior query, a scenario that needs `parentId` seems to defeat that entire purpose thus is out of scope.